From fef54359af73c63a95a35c3f1352d3d1df00c645 Mon Sep 17 00:00:00 2001
From: Sean Paul <seanpaul@chromium.org>
Date: Tue, 4 Mar 2014 17:37:24 -0500
Subject: [PATCH 11/15] drm: Add commit_state to drm_crtc_state

This patch adds a new bit to the crtc state which allows us to preclude
crtcs from being committed.

BUG=chromium:336809
TEST=Tested on pit, no regressions noted

Change-Id: I2631cc97e550069dc6dc17a0b041bc6af4cc3c2e
Signed-off-by: Sean Paul <seanpaul@chromium.org>
---
 drivers/gpu/drm/drm_atomic_helper.c | 5 +++++
 drivers/gpu/drm/drm_crtc.c          | 1 +
 include/drm/drm_crtc.h              | 1 +
 3 files changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/drm_atomic_helper.c b/drivers/gpu/drm/drm_atomic_helper.c
index 2386ab1..6d53af8 100644
--- a/drivers/gpu/drm/drm_atomic_helper.c
+++ b/drivers/gpu/drm/drm_atomic_helper.c
@@ -485,6 +485,7 @@ void drm_atomic_helper_init_crtc_state(struct drm_crtc *crtc,
 	/* this should never happen.. but make sure! */
 	WARN_ON(cstate->event);
 	cstate->event = NULL;
+	cstate->commit_state = false;
 }
 EXPORT_SYMBOL(drm_atomic_helper_init_crtc_state);
 
@@ -541,6 +542,7 @@ swap_crtc_state(struct drm_crtc *crtc, struct drm_atomic_helper_state *a)
 	cstate->set_config = false;
 	cstate->new_fb = false;
 	cstate->connectors_change = false;
+	cstate->commit_state = false;
 
 	swap(crtc->state, a->cstates[crtc->id]);
 	crtc->base.propvals = &crtc->state->propvals;
@@ -631,6 +633,9 @@ drm_atomic_helper_commit_crtc_state(struct drm_crtc *crtc,
 	struct drm_atomic_helper_state *a = cstate->state;
 	int ret = -EINVAL;
 
+	if (!cstate->commit_state)
+		return 0;
+
 	if (cstate->set_config)
 		return set_config(crtc, cstate);
 
diff --git a/drivers/gpu/drm/drm_crtc.c b/drivers/gpu/drm/drm_crtc.c
index 4834dd7..c8b2c54 100644
--- a/drivers/gpu/drm/drm_crtc.c
+++ b/drivers/gpu/drm/drm_crtc.c
@@ -1003,6 +1003,7 @@ int drm_crtc_set_property(struct drm_crtc *crtc,
 		return -EINVAL;
 	}
 
+	state->commit_state = true;
 	return 0;
 }
 EXPORT_SYMBOL(drm_crtc_set_property);
diff --git a/include/drm/drm_crtc.h b/include/drm/drm_crtc.h
index b81a64c..043ca1e 100644
--- a/include/drm/drm_crtc.h
+++ b/include/drm/drm_crtc.h
@@ -482,6 +482,7 @@ struct drm_crtc_state {
 	bool set_config        : 1;
 	bool new_fb            : 1;
 	bool connectors_change : 1;
+	bool commit_state      : 1;
 
 	uint8_t num_connector_ids;
 	uint32_t *connector_ids;
-- 
1.8.3.2

